src/aggregate/reducers/to_list.c

```diff
@@ -6,75 +6,70 @@
 
 #include <aggregate/reducer.h>
 
-typedef struct {
-  TrieMap *values;
-  const RLookupKey *srckey;
-} tolistCtx;
+static uint64_t hashFunction_RSValue(const void *key) {
+  return RSValue_Hash(key, 0);
+}
+static void *dup_RSValue(void *p, const void *key) {
+  return RSValue_IncrRef((RSValue *)key);
+}
+static int compare_RSValue(void *privdata, const void *key1, const void *key2) {
+  return RSValue_Equal(key1, key2, NULL);
+}
+static void destructor_RSValue(void *privdata, void *val) {
+  RSValue *v = val;
+  RSValue_Decref(v);
+}
+
+static dictType RSValueSet = {
+  .hashFunction = hashFunction_RSValue,
+  .keyDup = dup_RSValue,
+  .valDup = NULL,
+  .keyCompare = compare_RSValue,
+  .keyDestructor = destructor_RSValue,
+  .valDestructor = NULL,
+};
 
 static void *tolistNewInstance(Reducer *rbase) {
-  tolistCtx *ctx = Reducer_BlkAlloc(rbase, sizeof(*ctx), 100 * sizeof(*ctx));
-  ctx->values = NewTrieMap();
-  ctx->srckey = rbase->srckey;
-  return ctx;
+  dict *values = dictCreate(&RSValueSet, NULL);
```

Comment:
why the change from trie to dict? I think this is the most important question in this PR

Reply:
There was no good reason to abuse the triemap (designed for strings) to store the `uint64_t` hash that was calculated manually and then passed as a string. Also later iterating over the trie with an empty string so all the results will be found. Moving to a dictionary simplifies the flow (calling `dictAdd` instead of calculating the hash, checking if it already exists, and only then inserting the value). The dictionary handles all of the logic of taking references to new values, freeing them in the end, calculating the hash, and so on. Also iterating over it makes more sense.